src/channels/discord/interaction_handler.py

```
@@ -0,0 +1,1044 @@
+    SELECT_MENU = "select_menu"
+    MODAL_SUBMIT = "modal_submit"
+    AUTOCOMPLETE = "autocomplete"
+
+
+class DiscordInteractionHandler:
+    """
+    Handles Discord interactions including slash commands, buttons, and UI components
+    """
+    
+    def __init__(self, config: InstanceConfig, message_router: MessageRouter, 
+                 agent_interface: AgentInterface, cache_manager: CacheManager):
```

Ruff (0.12.2)

Undefined name `AgentInterface`

(F821)